src/app/filters/filter-fields.component.html

```diff
@@ -34,11 +35,38 @@
           <li *ngFor="let query of currentField.queries"
               [ngClass]="{'selected': query.value === currentValue}">
             <a role="menuitem" tabindex="-1" (click)="selectValue(query)">
+              <span *ngIf="query?.iconClass" class="fa {{query?.iconClass}}"></span>
+              <img class="avatar" *ngIf="query?.imageUrl" [attr.src]="query?.imageUrl" />
               {{query.value}}
             </a>
           </li>
         </ul>
       </div>
     </div>
+    <div *ngIf="currentField.type === 'typeahead'">
```

Comment:
Filtered list is needed to show the filtered queries by type ahead. Filtered list is nothing but a clone of **this.currentField.queries**, so should be fine. Every time you change the field it gets updated.